lib/matplotlib/backend_bases.py

```diff
@@ -1595,6 +1595,73 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    def _repr_html_(self):
+        if not self.figure.axes and not self.figure.lines:
+            return
```

Comment:
What if the figure only contains a Rectangle an arrow and a Text?

Reply:
True, I copied that from elsewhere. Is there a clean way to determine if the figure has changed at all?

Reply:
Check https://matplotlib.org/faq/howto_faq.html#check-whether-a-figure-is-empty 
However, maybe even completely empty figures should return some html? After all, you get what you ask for, right?

Reply:
What do you mean "changed"?

I agree with @ImportanceOfBeingErnest if we are going to have a repr, it should always show _something_ even if that something is an empty image.